software/bismark/align/main.nf

```diff
@@ -0,0 +1,65 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+process BISMARK_ALIGN {
+    tag "$meta.id"
+    label 'process_high'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::bismark==0.23.0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/bismark:0.23.0--0"
+    } else {
+        container "quay.io/biocontainers/bismark:0.23.0--0"
+    }
+
+    input:
+    tuple val(meta), path(reads)
+    path index
+
+    output:
+    tuple val(meta), path("*bam")       , emit: bam
+    tuple val(meta), path("*report.txt"), emit: report
+    tuple val(meta), path("*fq.gz")     , optional:true, emit: unmapped
```

Comment:
So you don't need to add any customisation here for  single- or paired-end reads e.g. [here](https://github.com/nf-core/modules/blob/4703a604431b48524a0e043a107eb76eb4674717/software/bowtie2/align/main.nf#L35)?

Reply:
`bismark` handles this. 
Excerpt from the [docs](https://github.com/FelixKrueger/Bismark/blob/60653293acc5ad1943865661fd17140322910aaa/Docs/README.md#output):

```
Written reads will appear as they did in the input, without any translation of quality values that may have taken place within Bowtie or Bismark. Paired-end reads will be written to two parallel files with _1 and _2 inserted in their filenames, i.e. unmapped_reads_1.fq.gz and unmapped_reads_2.fq.gz
```